lib/bigtable/table.js

```diff
@@ -0,0 +1,859 @@
+ *
+ * //-
+ * // Get the rows from your table as a readable object stream.
+ * //-
+ * table.getRows()
+ *   .on('error', console.error)
+ *   .on('data', function(row) {
+ *     // `row` is a Row object.
+ *   })
+ *   .on('end', function() {
+ *     // All rows retrieved.
+ *   });
+ *
+ * //-
+ * // If you anticipate many results, you can end a stream early to prevent
+ * // unnecessary processing.
+ * //-
+ * table.getRows()
+ *   .on('data', function(row) {
+ *     this.end();
+ *   });
+ */
+Table.prototype.getRows = function(options, callback) {
+  var self = this;
+
+  if (is.function(options)) {
+    callback = options;
+    options = {};
+  }
+
+  options = options || {};
+
+  var grpcOpts = {
+    service: 'BigtableService',
+    method: 'readRows',
+    stream: true
+  };
+
+  var reqOpts = {
+    tableName: this.name
+  };
+
+  if (options.key) {
+    reqOpts.rowKey = Mutation.convertToBytes(options.key);
+  } else if (options.start || options.end) {
+    reqOpts.rowRange = Table.formatRowRange_(options);
+  } else if (options.keys || options.ranges) {
+    reqOpts.rowSet = {};
+
+    if (options.keys) {
+      reqOpts.rowSet.rowKeys = options.keys.map(Mutation.convertToBytes);
+    }
+
+    if (options.ranges) {
+      reqOpts.rowSet.rowRanges = options.ranges.map(Table.formatRowRange_);
+    }
+  }
+
+  if (options.filter) {
+    reqOpts.filter = Filter.parse(options.filter);
+  }
+
+  if (options.interleave) {
+    reqOpts.allowRowInterleaving = options.interleave;
+  }
+
+  if (options.limit) {
+    reqOpts.numRowsLimit = options.limit;
+  }
+
+  var stream = pumpify.obj([
+    this.request(grpcOpts, reqOpts),
+    through.obj(function(rowData, enc, next) {
+      var row = self.row(Mutation.convertFromBytes(rowData.rowKey));
+
+      row.data = Row.formatChunks_(rowData.chunks);
+      next(null, row);
+    })
+  ]);
+
+  if (!is.function(callback)) {
+    return stream;
+  }
+
+  stream
+    .on('error', callback)
+    .pipe(concat(function(rows) {
+      callback(null, rows);
+    }));
+};
+
+/**
+ * Insert or update rows in your table.
+ *
+ * @param {object|object[]} entries - List of entries to be inserted.
+ *     See {module:bigtable/table#mutate}.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var entries = [
+ *  {
+ *     key: 'alincoln',
+ *     data: {
+ *       follows: {
+ *         gwashington: '1'
+ *       }
+ *     }
+ *   }
+ * ];
+ *
+ * table.insert(entries, function(err, apiResponse) {});
+ *
+ * //-
+ * // By default whenever you insert new data, the server will capture a
+ * // timestamp of when your data was inserted. It's possible to provide a
+ * // date object to be used instead.
+ * //-
+ * var entries = [
+ *   {
+ *     key: 'gwashington',
+ *     data: {
+ *       follows: {
+ *         jadams: {
+ *           value: '1',
```

Comment:
Somewhere there should be a discussion of data.  Bigtable is very much a binary thing, there should be somekind of discussion of dealing with that.   What do you do w/ 8byte longs?